### PR TITLE
Laboratory work #1, Vorontsova Anna - 22FPL2

### PR DESCRIPTION
**Pull request by vorontsann**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print ('Hello world!11')
```
Comment (artyomtugaryov) [inline main.py:1]: remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
unfortunately, nothing to review

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,7 +11,16 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
-
+    if isinstance(text, str) == False:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:14]: early return, please https://habr.com/ru/articles/348074/

File: main.py (hunk 1 above)
Comment (vorontsann) [reply main.py]: deleted

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:14]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -11,7 +11,16 @@ def tokenize(text: str) -> list[str] | None:
...
+        return None
+    else:
+        list_of_tokens = ""
+        for token in text:
+            if token.isalpha():
+                new_token = token.lower()
+                list_of_tokens += new_token
+        tokens = list(list_of_tokens)
+        return(tokens)
```
[the remark this answers is not in the document]
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -11,14 +11,36 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
 
+    list_of_tokens = ""
+    for token in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: try to simplify the function

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -11,14 +11,36 @@ def tokenize(text: str) -> list[str] | None:
...
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculates frequencies of given tokens
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list) or not all(isinstance(letter, str) for letter in tokens):
+        return None
+
+    list_of_tokens = ""
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:35]: for what?

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -28,6 +50,14 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+
+    tokens = tokenize(text)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:56]: you can set the value to the dict without creating a variable

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -37,6 +67,17 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
     """
+    if (len(predicted) != len(actual) or
+        not isinstance(predicted, list) or
+        not isinstance(actual, list)):
+        return None
+
+    sum_mse = 0
+    for i in range(len(actual)):
+        sum_mse += (actual[i]-predicted[i]) ** 2
+    mse = sum_mse/len(actual)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:78]: return without creating the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:35]: fixed (if i understood correctly)

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:56]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:78]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:18]: fixed

Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:18]: You can make list without intermediate variables. Use list comprehension

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (artyomtugaryov) [reply lab_1_classify_by_unigrams/main.py:33]: do not think so

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -37,6 +61,16 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+    if (len(predicted) != len(actual) or
+        not isinstance(predicted, list) or
+        not isinstance(actual, list)):
+        return None
+
+    sum_mse = 0
+    for i, act_value in enumerate(actual):
+        sum_mse += (act_value-predicted[i]) ** 2
+
+    return sum_mse/len(actual)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:73]: add spaces

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py:30]: now should be fixed 100% (i corrected it i don't know why nothing changed at first time :c )

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py]: got it now, thank you! fixed

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (vorontsann) [reply lab_1_classify_by_unigrams/main.py]: fixed

**Pull request closed without merging**